### ISSUE TITLE
Internal Viewer Code issues

### ISSUE THREAD

**Issue by oldium** [doublecmd/doublecmd#1461]
### Version

Double Commander
Version: 1.1.9 beta
Revision: 350
Commit: b220a50e9
Build date: 2024/01/14
Lazarus: 2.2.6.0
FPC: 3.2.2
Platform: x86_64-Win64-win32/win64
OS version: Windows 11 23H2 x86_64

### Description

The Code mode of Internal Viewer looks promising. I have found two issues so far:

* Dark mode - the highlight colors are sometimes dark too; it seems the Code mode does not have a dark mode - see screenshot.
* Viewing large files (like 2GB JSON file) - this takes considerable time to show (Double Commander gets unresponsive), so it could be good to limit the Code mode display by some configurable size limit

![image](https://github.com/doublecmd/doublecmd/assets/1260936/5a183c3b-8f7a-45d4-9fdb-dc34007eb372)

**Comment by alexx2000** [comment]
It should has a full dark mode support. Which highlighter do you use?

**Comment by oldium** [comment]
Thanks for the pointer. I am not using internal Editor (I use external one - Notepad++), so I have not touched any highlighting. But I checked how it behaves in my second account, which I use rarely (here I update Double Commander less often). On my second account the colors look good. So I compared AppData/Roaming/doublecmd and found lot of differences in `colors.json` - maybe some relics from older version (I have been using Double Commander snapshot for years). Is it safe to delete the file to get the defaults or how to proceed?

**Comment by alexx2000** [comment]
If you does not have a custom color settings then you can remove `colors.json`, otherwise you can remove `Highlightes/UniHighlighters` sections from `colors.json`. Both actions should be done with closed DC.

**Comment by oldium** [comment]
Deleted `colors.json` and now it looks good. Thanks. So the only issue now is support for opening really large files.

![image](https://github.com/doublecmd/doublecmd/assets/1260936/2d7af1fe-e4ea-4f31-82e1-deaa1da8d33e)

**Comment by oldium** [comment]
Should I create a separate issue for the viewing of large files? The color issue is now fixed for me, so this issue could be closed afterwards.

**Comment by oldium** [comment]
Yes, I know. GitHub knows only Issues. But I asked about something else - whether you want to keep the “feature request” separately or whether you are fine that it is mentioned here together with the color problem, which is fixed for me.
